### ISSUE TITLE
Cop idea: Implicit return in predicate method

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13063]
Ruby returns the value of the last evaluated expression. Predicate methods should return `true`/`false` but implicit returns can result in `nil` being the result.

## Expected behavior

Offense for the following

### case/when
```rb
# bad
def foo?
  case foo
  when bar
     true
  end
end

# good
def foo?
  case foo
  when bar
     true
  else
    false
  end
end
```

### inline if/unless

```rb
# bad
def foo?
  false if bar?
end


# good
def foo?
  return false if bar?

  true
end
```

### multiline if/unless

```rb
# bad
def foo?
  if bar?
    baz
    true
  end
end


# good
def foo?
  if bar?
    baz
    true
  else
    false
  end
end
```

**Comment by koic** [comment]
`Style/ReturnNilInPredicateMethodDefinition` is intended to check only the `return` keyword. So, this proposal doesn't align with the role of `Style/ReturnNilInPredicateMethodDefinition`. Checking the last evaluated value should be the role of a different cop.

**Comment by Earlopain** [comment]
I don't think I agree with your opinion. While the cop may only check for explicit returns right now, the effect is the same: a predicate method returns nil with this code.

**Comment by koic** [comment]
I understand what you are trying to convey, but as the designer of this cop, I can say that each cop should be simple and should not be assigned more roles than necessary. This cop is specifically focused on the style of `return` (not the last evaluated value) , and its cop name is intentionally reflective of that purpose.
https://docs.rubocop.org/rubocop/1.65/cops_style.html#stylereturnnilinpredicatemethoddefinition

**Comment by Earlopain** [comment]
Ok, I merged this with the other one and made it a feature request instead.

**Comment by dvandersluis** [comment]
I didn't realize this issue existed but I opened #13305 to handle one implicit case.
